packages/next-swc/crates/next-api/src/pages.rs

```diff
@@ -619,16 +496,136 @@ impl PageApiEndpoint {
 
         Ok(api_entry_chunk)
     }
+
+    #[turbo_tasks::function]
+    async fn pages_manifest(
+        self: Vc<Self>,
+        entry_chunk: Vc<Box<dyn OutputAsset>>,
+    ) -> Result<Vc<Box<dyn OutputAsset>>> {
+        let this = self.await?;
+        let node_root = this.pages_project.project().node_root();
+        let chunk_path = entry_chunk.ident().path().await?;
+
+        let asset_path = node_root
+            .join("server".to_string())
+            .await?
+            .get_path_to(&chunk_path)
+            .context("ssr chunk entry path must be inside the node root")?;
+
+        let pages_manifest = PagesManifest {
+            pages: [(this.pathname.await?.clone_value(), asset_path.to_string())]
+                .into_iter()
+                .collect(),
+        };
+        Ok(Vc::upcast(RawOutput::new(Vc::upcast(VirtualSource::new(
+            node_root.join("server/pages-manifest.json".to_string()),
```

Comment:
I don't think it's a good idea to write that to this global filename. We should write it to `server/{route}/pages-manifest.json` and let next.js take care of globalizing it

Reply:
I'll do the same for app manifests.